File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, n: int) -> int:
        revNum = 0
        sign = 1
        if n < 0:
            sign = -1
            n *= -1
        while n > 0:
            lastDigit = n % 10
            revNum = (revNum * 10) + lastDigit
            if revNum > 2**31 - 1:
                return 0
            n = n // 10

        if sign == -1:
            return -1 * revNum
        
        return revNum

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int revNum = 0;
        int dummy = x;

        while (dummy != 0) {
            int digit = dummy % 10;

            if ((revNum > Integer.MAX_VALUE / 10) || (revNum == Integer.MAX_VALUE && digit > 7)) {
                return 0;
            }

            if ((revNum < Integer.MIN_VALUE / 10) || (revNum == Integer.MIN_VALUE && digit < -8)) {
                return 0;
            }

            revNum = (revNum * 10) + digit;
            dummy = dummy / 10;
        }
        return revNum;
    }
}